### DISCUSSION TITLE
Model outputs zero loss, but RMSE values decrease over time

### DISCUSSION THREAD

**Discussion by tulga-rdn** [ACEsuit/mace#919] [Q&A] [answered]
Hello! I am using MACE on my datasets, but I have a small problem that the losses outputted by the model are zero:
```
2025-04-09 18:52:59.242 INFO: Initial: head: default, loss=0.00000000, RMSE_E_per_atom=    8.76 meV, RMSE_F=   95.30 meV / A
2025-04-09 18:53:20.944 INFO: Epoch 0: head: default, loss=0.00000000, RMSE_E_per_atom=    8.78 meV, RMSE_F=   94.57 meV / A
2025-04-09 18:53:39.072 INFO: Epoch 1: head: default, loss=0.00000000, RMSE_E_per_atom=    8.80 meV, RMSE_F=   93.85 meV / A
2025-04-09 18:53:57.167 INFO: Epoch 2: head: default, loss=0.00000000, RMSE_E_per_atom=    8.83 meV, RMSE_F=   93.12 meV / A
2025-04-09 18:54:15.309 INFO: Epoch 3: head: default, loss=0.00000000, RMSE_E_per_atom=    8.85 meV, RMSE_F=   92.41 meV / A
2025-04-09 18:54:33.460 INFO: Epoch 4: head: default, loss=0.00000000, RMSE_E_per_atom=    8.86 meV, RMSE_F=   91.70 meV / A
...
2025-04-09 23:33:12.286 INFO: Epoch 962: head: default, loss=0.00000000, RMSE_E_per_atom=    4.09 meV, RMSE_F=   18.47 meV / A
2025-04-09 23:33:29.658 INFO: Epoch 963: head: default, loss=0.00000000, RMSE_E_per_atom=    4.09 meV, RMSE_F=   18.46 meV / A
```
And then it stops at epoch 2048 saying `Stopping optimization after 2048 epochs without improvement` However, based on the RMSE values, the model seems to be learning from data, so the problem is probably in how I set up my training script:
```
mace_run_train \
    --E0s="average" \
    --device=cuda \
    --batch_size=32 \
    --max_num_epochs=300000000000000 \
    --model="MACE" \
    --hidden_irreps='128x0e + 128x1o' \
    --max_L=1 \
    --r_max=5.0 \
    --amsgrad \
    --energy_weight=1 \
    --forces_weight=1 \
    --lr=1e-5 \
    --eval_interval=1 \
    --keep_checkpoints \
    --save_all_checkpoints \
    --enable_cueq=True \
```
It's also possible that the loss values are too small to be printed, but it seems that it's truly zero for the model as well, as it says that all of the epochs were without improvement. Am I missing something in the training script or should I just, for example, override the patience flag to a higher number? Thanks in advance!

**Comment by vue1999** [comment]
Could you please share the log file? It’s difficult to verify whether the file and the energy/force keys are being read correctly, as those details aren’t visible in the training script.

**Reply by tulga-rdn** [reply]
it seems that the .xyz files are read correctly:
[cchain_1sttry_run-20210792.log](https://github.com/user-attachments/files/19757692/cchain_1sttry_run-20210792.log)

**Reply by vue1999** [reply, marked as answer]
Seems that you are not loading energy values: '[9018 configs, 0 energy, 9018 forces, 9018 stresses]'.
Can you please check if your energy_keys (and force_keys) are correct.
It is also recommended that you calculate your own E0 values.

**Reply by tulga-rdn** [reply]
Thanks, it worked! For reference, what's the default energy key?

**Reply by vue1999** [reply]
Great! The default keys are: REF_energy, REF_forces, REF_stress
